lib/internal/Magento/Framework/Mail/Message.php

```diff
@@ -34,6 +34,23 @@ public function setBody($body)
     {
         return $this->messageType == self::TYPE_TEXT ? $this->setBodyText($body) : $this->setBodyHtml($body);
     }
+    
+    /**
+     * Set from address
+     *
+     * @param string|array $email
+     * @return $this
+     */
+    public function setFrom($email)
+    {
+        $name = null;
+        if (is_array($email)) {
+            $name = array_keys($email)[0];
+            $name = !is_int($name) ? $name : null;
+            $email = array_values($email)[0];
```

Comment:
This is `O(n)` while can be `O(1)` quite easily.

Something like
```
foreach ($email as $key => $value) {
    $name = is_int($key) ? null : $key;
    $email = $value;
    break;
}
```
or maybe even 
```
foreach ($email as $name => $email) {
    break;
}
$name = !is_int($name) ? $name : null;
```
should do the trick 😉 

Unit test maybe?